cpp/models/abm/infection.h

```diff
@@ -87,53 +84,52 @@ class Infection
 public:
     /**
      * create an infection for a single person.
+     * @param virus virus type of the infection
      * @param start_date starting date of the infection
-     * @param virus_type virus type of the infection
+     * @param start_state starting infection state of the person, default susceptible. Only for initialization.
      */
-    Infection(TimePoint start_date, std::shared_ptr<Virus> virus)
-        : m_start_date(start_date)
-        , m_virus(virus)
-        , m_viral_load()
-    {
-        m_end_date = m_viral_load.determine_end_date(m_start_date);
-    };
+    Infection(std::shared_ptr<Virus> virus, const TimePoint& start_date, const InfectionState& start_state = InfectionState::Exposed, const bool detected = false);
 
     /**
      * get viral load at a given time
      * @param t time point of the querry
      * @return viral load at given time point
      */
-    double get_viral_load(const TimePoint t) const;
+    double get_viral_load(const TimePoint& t) const;
 
     /**
      * get infectivity at a given time
      * @param t time point of the querry
      * @return infectivity at given time point
      */
-    double get_infectivity(const TimePoint t) const;
+    double get_infectivity(const TimePoint& t) const;
 
     /**
      * get virus type
      * @return virus type of the infection
      */
-    VirusType get_virus_type() const;
+    std::shared_ptr<Virus> get_virus_type() const;
 
-    InfectionState get_infection_state(const TimePoint t) const;
-
-    void draw_infection_course(InfectionState start_state = InfectionState::Susceptible);
+    InfectionState get_start_infection_state() const;
```

Comment:
Why do we need that? We could call get_infection_state with t=t_min?

Reply:
Yes, I had some idea behind that but forgot... maybe it was related to not needing t as a parameter. I will delete it if we don't put any good use to it anywhere.